### PR TITLE
fix: terminate fetch on gc

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

**Comment by ronag** [comment]
@szmarczak

File: lib/fetch/index.js [hunk 1]
```diff
@@ -30,6 +30,20 @@ const { ReadableStream } = require('stream/web')
...
+  const next = []
+  for (const entry of registry) {
+    if (entry.ref.deref() === undefined) {
+      entry.abort()
+    } else {
+      next.push(entry)
+    }
+  }
+  registry = next
+}, 1e3).unref()
```
Comment (ronag) [inline lib/fetch/index.js:45]: We could use a finalizationRegistry as well but that would be a lot less deterministic and the user might queue up lot's of requests before GC would start up.

Comment (ronag) [reply lib/fetch/index.js:45]: The number of elements in the "registry" is limited by the number of active connections so it should normally be quite small.

Comment (ronag) [reply lib/fetch/index.js:45]: @mcollina wdyt? deterministic polling loop or undeterministic `FinalizationRegistry`?

Comment (mcollina) [reply lib/fetch/index.js:45]: The polling is unfeasible, it will keep processes active even with no actual traffic. Let's go with FinalizationRegistry.

Comment (ronag) [reply lib/fetch/index.js]: Fixed. PTAL.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by szmarczak** [comment]
> The user agent can terminate the fetch because the associated body is not observable.

> `window.promise = fetch("https://www.example.com/").then(res => res.headers)`

If this isn't done already, can you patch this?

**Comment by ronag** [comment]
> If this isn't done already, can you patch this?

I think that is done.

**Pull request merged**